### PR TITLE
Text indentation

### PR DESCRIPTION
**Pull request by maimonav**
Click below to open examples:
rich-content: https://rich-content-textIndent.surge.sh/
rich-content-storybook: https://rich-content-storybook-textIndent.surge.sh/

### CODE REVIEW

File: packages/editor-common/web/src/Utils/draftUtils.js [hunk 1]
```diff
@@ -537,6 +537,62 @@ export function setSelection(editorState, selection) {
...
+  const depth = block.getDepth();
+
+  if (isDirectionNext && depth === maxDepth) {
+    return editorState;
+  }
+
+  const withAdjustment = adjustBlockDepthForContentState(
+    content,
+    selection,
+    isDirectionNext ? 1 : -1,
```
Comment (shaulgo) [inline packages/editor-common/web/src/Utils/draftUtils.js:590]:
```suggestion
    direction ? 1 : -1,
```

File: packages/editor-common/web/src/index.js [hunk 2]
```diff
@@ -86,6 +86,8 @@ export {
   getPostContentSummary,
   createSelection,
   getBlockType,
+  onTab,
```
Comment (shaulgo) [inline packages/editor-common/web/src/index.js:89]: remove onTab

File: packages/editor-common/web/src/Utils/draftUtils.js [hunk 3]
```diff
@@ -537,6 +537,62 @@ export function setSelection(editorState, selection) {
...
+  });
+}
+
+export function onTab(event, editorState) {
+  event.preventDefault();
+  const isDirectionNext = !event.shiftKey;
+  return onIndent(isDirectionNext, editorState);
+}
+
+export function onIndent(isDirectionNext, editorState) {
```
[the remark this answers is not in the document]
Comment (shaulgo) [reply packages/editor-common/web/src/Utils/draftUtils.js:570]: rename onIndent to indentSelectedBlock

File: packages/plugin-indent/web/src/icons/NextIndentPluginIcon.svg [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="19" height="19" viewBox="0 0 19 19">
```
Comment (shaulgo) [inline packages/plugin-indent/web/src/icons/NextIndentPluginIcon.svg:1]: rename svg file

File: packages/plugin-indent/web/src/icons/PrevIndentPluginIcon.svg [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="19" height="19" viewBox="0 0 19 19">
```
Comment (shaulgo) [inline packages/plugin-indent/web/src/icons/PrevIndentPluginIcon.svg:1]: rename svg file

File: packages/plugin-indent/web/src/defaults.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import { INDENT_TYPE as type } from './types';
+export const DEFAULTS = Object.freeze({
+  type,
+  config: {},
+});
+
+//@colors is defined in 'ThemeGenerator.js'
```
Comment (shaulgo) [inline packages/plugin-indent/web/src/defaults.js:7]: try to remove. 
Also, you can remove the colors param and the comment for eslint

File: packages/plugin-indent/web/src/createIndentPlugin.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+  return createBasePlugin({
+    type: INDENT_TYPE,
+    toolbar: createToolbar({
+      isMobile,
+    }),
+    helpers,
+    settings,
+    t,
+    isMobile,
+    disableRightClick: config?.uiSettings?.disableRightClick,
```
Comment (shaulgo) [inline packages/plugin-indent/web/src/createIndentPlugin.js:18]: remove this line

File: packages/plugin-indent/web/src/toolbar/NextIndentButton.jsx [hunk 8]
```diff
@@ -0,0 +1,40 @@
+import React, { Component } from 'react';
```
Comment (shaulgo) [inline packages/plugin-indent/web/src/toolbar/NextIndentButton.jsx:1]: rename file

File: packages/plugin-indent/web/src/toolbar/NextIndentButton.jsx [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+import PropTypes from 'prop-types';
+import { InlineToolbarButton, onIndent } from 'wix-rich-content-editor-common';
+import { INDENT_TYPE } from '../types';
+import NextIndentPluginIcon from '../icons/NextIndentPluginIcon.svg';
+
+export default class NextIndentButton extends Component {
+  render() {
+    const { theme, isMobile, t, tabIndex, setEditorState, getEditorState, config } = this.props;
+    const icon =
+      config?.[INDENT_TYPE]?.toolbar?.icons?.InsertPluginButtonIcon || NextIndentPluginIcon;
```
Comment (shaulgo) [inline packages/plugin-indent/web/src/toolbar/NextIndentButton.jsx:11]: remove icon customization

File: packages/plugin-indent/web/src/toolbar/NextIndentButton.jsx [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    return (
+      <InlineToolbarButton
```
Comment (shaulgo) [inline packages/plugin-indent/web/src/toolbar/NextIndentButton.jsx:13]: try factory method for the buttons. If that doesn't work, put both buttons in one file

File: packages/plugin-indent/web/src/toolbar/createToolbar.js [hunk 11]
```diff
@@ -0,0 +1,24 @@
+import PrevIndentButton from './PrevIndentButton';
```
Comment (shaulgo) [inline packages/plugin-indent/web/src/toolbar/createToolbar.js:1]: rename

File: packages/plugin-indent/web/src/toolbar/createToolbar.js [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+import NextIndentButton from './NextIndentButton';
+
+export default function createToolbar({ isMobile }) {
+  return {
+    TextButtonMapper: () => ({
+      PrevIndent: {
+        component: PrevIndentButton,
+        isMobile,
+        position: {
+          desktop: 10,
```
Comment (shaulgo) [inline packages/plugin-indent/web/src/toolbar/createToolbar.js:11]: fix position group

File: packages/common/web/statics/locale/messages_en.json [hunk 13]
```diff
@@ -293,6 +293,8 @@
   "HeadingThreeButton_tooltip": "Heading 3",
   "NormalTextButton_tooltip": "Normal text",
   "TextCodeBlockButton_Tooltip": "Code snippet",
+  "PrevIndentButton_Tooltip": "Decrease indent",
```
[the remark this answers is not in the document]
Comment (maimonav) [reply packages/common/web/statics/locale/messages_en.json:296]: fixed.

File: packages/plugin-indent/web/src/createIndentPlugin.js (hunk 7 above)
Comment (maimonav) [reply packages/plugin-indent/web/src/createIndentPlugin.js:18]: fixed.

File: packages/plugin-indent/web/src/icons/NextIndentPluginIcon.svg (hunk 4 above)
Comment (maimonav) [reply packages/plugin-indent/web/src/icons/NextIndentPluginIcon.svg:1]: fixed.

File: packages/plugin-indent/web/src/icons/PrevIndentPluginIcon.svg (hunk 5 above)
Comment (maimonav) [reply packages/plugin-indent/web/src/icons/PrevIndentPluginIcon.svg:1]: fixed.

File: packages/plugin-indent/web/src/toolbar/NextIndentButton.jsx (hunk 8 above)
Comment (maimonav) [reply packages/plugin-indent/web/src/toolbar/NextIndentButton.jsx:1]: fixed.

File: packages/plugin-indent/web/src/toolbar/NextIndentButton.jsx (hunk 9 above)
Comment (maimonav) [reply packages/plugin-indent/web/src/toolbar/NextIndentButton.jsx:11]: fixed.

File: packages/plugin-indent/web/src/toolbar/createToolbar.js (hunk 11 above)
Comment (maimonav) [reply packages/plugin-indent/web/src/toolbar/createToolbar.js:1]: fixed.

File: packages/plugin-indent/web/src/toolbar/createToolbar.js (hunk 12 above)
Comment (maimonav) [reply packages/plugin-indent/web/src/toolbar/createToolbar.js:11]: fixed.

File: packages/common/web/statics/styles/draftDefault.rtlignore.scss [hunk 14]
```diff
@@ -11,6 +12,59 @@
...
+  .public-DraftStyleDefault-block-depth0.public-DraftStyleDefault-rtl{
+    margin-right: 0em !important;
+  }
+  
+  .public-DraftStyleDefault-block-depth1 .public-DraftStyleDefault-rtl,
+  .public-DraftStyleDefault-block-depth1.public-DraftStyleDefault-rtl{
+    margin-right: 1.5em !important;
+  }
+  
+  .public-DraftStyleDefault-block-depth2 .public-DraftStyleDefault-rtl,
```
[the remark this answers is not in the document]
Comment (maimonav) [reply packages/common/web/statics/styles/draftDefault.rtlignore.scss:50]: fixed. Thank you!!

File: packages/editor-common/web/src/Utils/draftUtils.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (maimonav) [reply packages/editor-common/web/src/Utils/draftUtils.js:570]: fixed.

File: packages/editor-common/web/src/Utils/draftUtils.js [hunk 15]
```diff
@@ -537,6 +537,62 @@ export function setSelection(editorState, selection) {
...
+  blockMap = blockMap.merge(blocks);
+
+  return contentState.merge({
+    blockMap,
+    selectionBefore: selectionState,
+    selectionAfter: selectionState,
+  });
+}
+
+export function onTab(event, editorState) {
```
[the remark this answers is not in the document]
Comment (maimonav) [reply packages/editor-common/web/src/Utils/draftUtils.js:564]: fixed.

File: packages/editor-common/web/src/Utils/draftUtils.js (hunk 1 above)
Comment (maimonav) [reply packages/editor-common/web/src/Utils/draftUtils.js:590]: fixed.

File: packages/editor-common/web/src/index.js (hunk 2 above)
Comment (maimonav) [reply packages/editor-common/web/src/index.js:89]: fixed.

File: packages/plugin-indent/web/src/toolbar/NextIndentButton.jsx (hunk 10 above)
Comment (maimonav) [reply packages/plugin-indent/web/src/toolbar/NextIndentButton.jsx]: fixed.

File: examples/main/shared/editor/EditorPlugins.jsx [hunk 16]
```diff
@@ -460,6 +463,13 @@ const config = {
     //   },
     // },
   },
+  [INDENT_TYPE]: {
```
Comment (shaulgo) [inline examples/main/shared/editor/EditorPlugins.jsx:466]: remove config

Comment (maimonav) [reply examples/main/shared/editor/EditorPlugins.jsx:466]: fixed.

File: packages/editor/web/src/RichContentEditor/handleKeyCommand.js [hunk 17]
```diff
@@ -21,12 +22,16 @@ export default (updateEditorState, customHandlers, blockType) => (command, edito
...
-    if (isTab(command) && isList(blockType)) {
-      // eslint-disable-next-line no-restricted-globals
-      newState = RichUtils.onTab(event, editorState, 2);
-    } else if (isText(blockType)) {
-      newState = insertString(editorState, CHARACTERS.TAB);
-    } else if (!isCodeBlock(blockType)) {
+    if (isTab(command)) {
+      if (isList(blockType)) {
+        // eslint-disable-next-line no-restricted-globals
+        newState = indentSelectedBlock(editorState, !event.shiftKey ? 1 : -1);
```
Comment (shaulgo) [inline packages/editor/web/src/RichContentEditor/handleKeyCommand.js:28]: lets do `const direction = !event.shiftKey ? 1 : -1`

File: packages/plugin-indent/web/src/toolbar/IndentButtons.jsx [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+      dataHook={dataHook}
+      tabIndex={tabIndex}
+      icon={icon}
+    />
+  );
+}
+
+export function DecreaseIndentButton(props) {
+  return indentButton(
+    props,
```
Comment (shaulgo) [inline packages/plugin-indent/web/src/toolbar/IndentButtons.jsx:30]: move all parama in to object. {...props, direction:...}

File: packages/editor/web/src/RichContentEditor/handleKeyCommand.js (hunk 17 above)
Comment (maimonav) [reply packages/editor/web/src/RichContentEditor/handleKeyCommand.js:28]: fixed.

File: packages/plugin-indent/web/src/toolbar/IndentButtons.jsx (hunk 18 above)
Comment (maimonav) [reply packages/plugin-indent/web/src/toolbar/IndentButtons.jsx:30]: fixed.

**Pull request merged**